### DISCUSSION TITLE
Feasible to make a NEXTRA doc site available & usable offline?

### DISCUSSION THREAD

**Discussion by mi-hol** [shuding/nextra#2625] [Q&A]
**Use case:**

- Knowledge-base for automotive troubleshooting developed with NEXTRA docs-template [boc-kb-v2.vercel.app](https://boc-kb-v2.vercel.app/)

**Requirement:**

- make it usable without internet connection

Is this feasible (i.e. via PWA offline feature)?

**Comment by mi-hol** [comment]
@dimaMachina as the most knowledgeable dev, maybe you'd have an answer?

**Reply by dimaMachina** [reply]
I think it could be done in Nextra 4 with app router support and by following Next.js docs about PWA integration

More info about Nextra 4 is written [here](https://github.com/shuding/nextra/issues/2600#issuecomment-2385107206)

**Reply by mi-hol** [reply]
unfortunately [this discussion](https://github.com/serwist/serwist/discussions/205) brings up an issue with Next 14's   `useRouter from next/navigation` the author could not overcome.

I've no understanding of Next and fear my limited skills will prevent me from being successful, unless I could get technical support from you to overcome them. 
Would you see "allow Nextra doc site to be used completely offline" as an important feature to justify your investment of time?

**Reply by mi-hol** [reply]
> Next.js docs about PWA integration

unfortunately the [doc I found here](https://github.com/serwist/serwist/blob/main/examples/next-basic/README.md) regarding  OFFLINE support is extremely brief and doesn't really ring a bell with me

**Reply by mi-hol** [reply]
With Nextra 4 stabilized would you consider PWA as a new feature [(similar to docusaurus/plugin-pwa)](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-pwa) worth supporting?
